pkg/apiserver/rest/usecase/helm.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright 2021 The KubeVela Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package usecase
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"time"
+
+	"helm.sh/helm/v3/pkg/repo"
+
+	"github.com/oam-dev/kubevela/pkg/apiserver/rest/utils"
+
+	"github.com/oam-dev/kubevela/pkg/utils/helm"
+)
+
+const (
+	versionPatten = "repoUrl: %s, chart: %s"
+	valuePatten   = "repoUrl: %s, chart: %s, version: %s"
+)
+
+// NewHelmUsecase return a helmHandler
+func NewHelmUsecase() HelmHandler {
+	return defaultHelmHandler{
+		repoCache:    map[string]*utils.MemoryCache{},
+		versionCache: map[string]*utils.MemoryCache{},
+		valuesCache:  map[string]*utils.MemoryCache{},
+	}
+}
+
+// HelmHandler responsible handle helm related interface
+type HelmHandler interface {
+	ListChartNames(ctx context.Context, url string) ([]string, error)
+	ListChartVersions(ctx context.Context, url string, chartName string) (repo.ChartVersions, error)
+	GetChartValues(ctx context.Context, url string, chartName string, version string) (map[string]interface{}, error)
+}
+
+type defaultHelmHandler struct {
+	repoCache    map[string]*utils.MemoryCache
+	versionCache map[string]*utils.MemoryCache
+	valuesCache  map[string]*utils.MemoryCache
+}
+
+func (d defaultHelmHandler) ListChartNames(ctx context.Context, url string) ([]string, error) {
+	if m, ok := d.repoCache[url]; ok && !m.IsExpired() {
+		return m.GetData().([]string), nil
+	}
+	helper := &helm.Helper{}
+	charts, err := helper.ListChartsFromRepo(url)
+	if err != nil {
+		return nil, err
+	}
+	d.repoCache[url] = utils.NewMemoryCache(charts, 10*time.Minute)
+	return charts, nil
+}
+
+func (d defaultHelmHandler) ListChartVersions(ctx context.Context, url string, chartName string) (repo.ChartVersions, error) {
+	if m, ok := d.versionCache[fmt.Sprintf(versionPatten, url, chartName)]; ok && !m.IsExpired() {
+		return m.GetData().(repo.ChartVersions), nil
+	}
+	helper := &helm.Helper{}
+	chartVersions, err := helper.ListVersions(url, chartName)
+	if err != nil {
+		return nil, err
+	}
+	// helm version updating is more frequent， so set 1min expired time
+	d.versionCache[fmt.Sprintf(versionPatten, url, chartName)] = utils.NewMemoryCache(chartVersions, 1*time.Minute)
```

Comment:
why this cache time is not like the others.

Reply:
As the comment, Creating new version for some helm charts is more frequent operation. If user upload a new version, must wait ten minutes can find it，it's a bad ux.

Reply:
Charts and Versions should be the same data source, all from `index.yaml`?

Reply:
yes

Reply:
so, it should be the same cache or the same cache duration

Reply:
make sense. Set them with 3 minute samely.